### ISSUE TITLE
Blaze: WordPress logo is displayed before populating the posts lists

### ISSUE THREAD

**Issue by tiagomar** [wordpress-mobile/WordPress-Android#18093]
Registering in case we decide to move on with web view approach for Blaze.

### Expected behavior
WordPress logo is not displayed while setting up a campaign.

### Actual behavior
WordPress logo is displayed before populating the posts lists, possibly form Claypso.

| **Video** | **Screenshot** |
| - | - |
| <video height="640" src="https://user-images.githubusercontent.com/42008628/224837594-c40bb7e0-4d6d-4a47-a105-d8fa7ac27729.mp4"> | <img height="632" src="https://user-images.githubusercontent.com/42008628/224838437-f560616b-45ce-4b16-b3ca-fb67b0e8c393.png"> |

### Steps to reproduce the behavior
0. Make sure that Blaze is enabled via the debug menu
1. Select a site that has Blaze enabled.
2. Tap the Blaze Card in Dashboard.
3. While loading the posts lists, notice that the WordPress logo is displayed.

**Comment by hypest** [comment]
I wonder if there's a "fully loaded" Calypso event we can catch, and use it to hide a native "loading" screen. We do [something similar with the Unsupported Block Editor](https://github.com/WordPress/gutenberg/blob/0e1305760947ca6876f7a45c32ce7e7833c7cac6/packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergWebViewActivity.java#L321), where we show a native custom loading screen while we wait for the webview to finish loading Gutenberg. cc @zwarm as you might have already considered it, not sure.

**Comment by zwarm** [comment]
@hypest 

> I wonder if there's a "fully loaded" Calypso event we can catch

Not that I have encountered. In fact, the starting/loading/finished URL remain the same for the post list: `https://wordpress.com/advertising/zieglerkirby.wordpress.com?blazepress-widget=post-285&_source=posts_list`. With that URL, there is no indication that the view is loading or has completed loading. 

@enejb  - Is there any insight you can add here? And of course, please correct me if I have misspoken.

**Comment by zwarm** [comment]
@hypest @enejb - FYI: The same "W" can be seen loading within iOS as well :(

**Comment by hypest** [comment]
> @hypest @enejb - FYI: The same "W" can be seen loading within iOS as well :(

Yeap, this is coming from Calypso so it's expected to be visible in both Android and iOS. I've only commented on the Android ticket because I have some insight on how we have the UBE webview transition implemented on Android.

**Comment by zwarm** [comment]
>  I've only commented on the Android ticket because I have some insight on how we have the UBE webview transition implemented on Android.

@hypest  Yep, I tried delaying the webview being shown until 1-5 seconds after load, but it didn't change the behavior. The "W" is still visible.
```
@SuppressLint("SetJavaScriptEnabled")
    @Composable
    fun BlazeWebViewContent(
        model: BlazeWebViewContentUiState,
        delayMillis: Long = 5000
    ) {
        val webView = WebView(requireContext()).apply {
            layoutParams = ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
            )
            scrollBarStyle = View.SCROLLBARS_INSIDE_OVERLAY
            settings.userAgentString = model.userAgent
            settings.javaScriptEnabled = model.enableJavascript
            settings.domStorageEnabled = model.enableDomStorage
            webViewClient = BlazeWebViewClient(this@BlazeWebViewFragment)
            chromeClient = WPWebChromeClientWithFileChooser(
                activity,
                this,
                R.drawable.media_movieclip,
                null,
                this@BlazeWebViewFragment
            )
            webChromeClient = chromeClient
            postUrl(WPCOM_LOGIN_URL, model.addressToLoad.toByteArray())
        }

        Box {
            LaunchedEffect(Unit) {
                delay(delayMillis)
            }
            AndroidView({ webView })
        }
    }

```

This should have created the WebView and started loading but delay it for 5 seconds before finally showing it on the screen.

**Comment by hypest** [comment]
> This should have created the WebView and started loading but delay it for 5 seconds before finally showing it on the screen.

Aha, not sure if the webview has some optimization to postpone the network work until visible(?). Want to try an overlay stacked on top of the webview and self-dismiss it after 5secs?

**Comment by zwarm** [comment]
> Want to try an overlay stacked on top of the webview and self-dismiss it after 5secs?

@hypest Interesting approach, I could give it a whirl in a little bit. This may have to do with our need to "post" the url vs. load it. The login action needs to happen before the page loads. I can see the same thing if I take the URL and use a browser on my desktop.

**Comment by enejb** [comment]
We can hide the W in the app from showing up by updating calypso.

**Comment by zwarm** [comment]
> We can hide the W in the app from showing up by updating calypso.

That would be awesome @enejb!! let me know how I can help.

**Comment by zwarm** [comment]
Took a very quick stab at creating a POC that shows a loading indicator. Is this what you were thinking  @hypest ? I'm not comfortable rushing this out without eyes on it, but it _could be_ an alternative if @enejb gets stuck. Although it doesn't solve the iOS part. :(

https://user-images.githubusercontent.com/506707/225078730-935872a5-58ac-4721-9af9-4ae5db820ae8.mp4

**Comment by enejb** [comment]
Here is the calypso PR. https://github.com/Automattic/wp-calypso/pull/74439 that changes the logo from W to the jetpack one.

**Comment by enejb** [comment]
I am closing this since it should be fixed with https://github.com/Automattic/wp-calypso/pull/74439
